Mathlib/Topology/ContinuousMap/StarOrdered.lean

```diff
@@ -3,63 +3,50 @@ Copyright (c) 2024 Jireh Loreaux. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Jireh Loreaux
 -/
-import Mathlib.Analysis.Complex.Basic
-import Mathlib.Data.Real.StarOrdered
+import Mathlib.Algebra.Order.Star.Basic
 import Mathlib.Topology.ContinuousMap.Algebra
 import Mathlib.Topology.ContinuousMap.ContinuousMapZero
 
-/-! # Continuous functions as a star-ordered ring -/
+/-! # Continuous functions as a star-ordered ring
 
-open scoped NNReal
+The type class `ContinuousSqrt` gives a sufficient condition on `R` to make `C(α, R)`
+and `C(α, R)₀` into a `StarOrderedRing` for any topological space `α`, thereby providing a means
+by which we can ensure `C(α, R)` has this property. This condiiton is satisfied
+by `ℝ≥0`, `ℝ`, and `ℂ`, and the instances can be found in the file
+`Topology.ContinuousMap.ContinuousSqrt`.
+-/
+
+/-- A type class encoding the property that there is a continuous square root function on
+nonnegative elements. This holds for `ℝ≥0`, `ℝ` and `ℂ` (as well as any C⋆-algebra), and this
+allows us to derive an instance of `StarOrderedRing C(α, R)` under appropriate hypotheses.
+In order for this to work on `ℝ≥0`, we actually must force our square root function to be defined
+on and well-behaved for pairs `x : R × R` with `x.1 ≤ x.2`. -/
+class ContinuousSqrt (R : Type*) [LE R] [NonUnitalSemiring R] [TopologicalSpace R] where
+  /-- `sqrt (a, b)` returns a value `s` such that `b = a + s * s` when `a ≤ b`. -/
+  protected sqrt : R × R → R -- should this really be data? We'll get diamonds for `ℂ`.
```

Comment:
What diamond are you thinking of?

Reply:
You can write an instance of this class for C*-algebras too, but the square root will go via the continuous functional calculus. But then `Complex` will have two non-propeq instances of this class. It's probably not a big deal because we essentially only care about this class to get the `StarOrderedRing` instance on the continuous functions, which is a `Prop`, and so there's virtually no reason to refer to this `sqrt` function.

Reply:
If you want to enforce this, you could make `ContinuousSqrt : Prop`

Reply:
I know, but that's a bit gross, and I don't think the existence of diamonds here is actually going to cause issues, but I'll add a warning.